README.md

```diff
@@ -1,43 +1,58 @@
-  - `LLMEvaluator` component outputs the assessment results in percentage. With this, we could decide the coverage rate. 
-- Continuous pipeline
-  - If the converage rate from the `LLMEvaluator` is still below a certain threshold, we could ask `SynthDataGen` component to generate more of synthetic data.
-
-## Todos
-
-- [ ] Notebook to define coverage dataset.
-- [ ] Notebook to define `SynthDataGen` component's functionality.
-- [ ] Notebook to define `LLMEvaluator` component's functionality.
-- [ ] Implement `SynthDataGen` and `LLMEvaluator` as custom TFX component.
-- [ ] Notebook to define a portion of TFX pipeline with the custom component to fine-tune Gemma 2B/7B models.
-- [ ] Implement a whole TFX pipeline including automatically exporting models and application to Hugging Face Hub.
\ No newline at end of file
+## Overview
+
+This project comes with the toolset of batch inference, evaluation, and synthetic data generation. Each tool can be run independently, but they could be hooked up to form a pipeline. It is your choice how to use them.
+
+The prerequisite to run these toolset is to have a dataset consisting of desired `(prompt, response)` pairs. The exact format of the dataset could be found [here](https://huggingface.co/datasets/sayakpaul/no_robots_only_coding). The `prompt` is the input to the small size LLM to generate output. Then, `prompt`, `response`, and the `generated output` are going to be used to evaluate the fine-tuned small size LLM. The main idea is to make small size LLM to output as much as similar to the given response.
+
+### Batch inference
+
+Batch inference lets fine-tuned LLM to generate text and push the results on the Hugging Face Dataset repository. 
+
+To perform this you need to run the following commands in terminal:
+
+```console
+# HF_TOKEN is required to access gated model repository 
+# and push the resulting outputs to the Hugging Face Hub.
+$ export HF_TOKEN=<YOUR-HUGGINGFACE-ACCESS-TOKEN>
+
+# All parameters defined in the config/batch_inference.yaml file
+# could be manually inputted as CLI arguments (arg names are the same)
+$ python batch_inference.py --from-config config/batch_inference.yaml
+```
+
+Then, the resulting outputs will be pushed to Hugging Face Dataset repository in the following structure:
+
+| column names | instructions |  target_responses |  candidate_responses  | model_id  |  model_sha |
+|---|---|---|---|---|---|
+| descriptions | the input prompts | desired outputs |  model generated outputs  |  model id that generated outputs  |  the version of the model |
+
+### Evaluation
+
+Evaluation evaluates the generated text from fine-tuned LLM with the help of service LLM. The evaluation criteria is the similarity and quality by comparing to the given desired outputs.
+
+(Instruction WIP)
```

Comment:
While they're WIP it could make sense to redirect the users to the notebooks and scripts related to them.